### ISSUE TITLE
Should respect XDG for user hooks

### ISSUE THREAD

**Issue by jasonkarns** [git-hooks/git-hooks#36]
https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html

The XDG basedir spec outlines the guidelines for applications and tools to store configuration, data, and cache files in a consistent, and user-configurable manner. This specification is aimed at Linux distributions but is 100% compatible with mac and windows as all the variables are derived and default from `$HOME` (which exists and is specified for both mac and windows).

git itself respects XDG (it respects the config file at `$XDG_CONFIG_HOME/git/config`)

TL;DR: git-hooks should look for user hooks under `${XDG_CONFIG_HOME:-$HOME/.config}/githooks/` and not `~/.githooks`

**Comment by CatTail** [comment]
Hi @jasonkarns, I agree that follow XDG spec to put configuration under `$XDG_CONFIG_HOME/git/config` make more sense, can you help to make these changes to this project.
